tests/unit/test_newsrec_utils.py

```diff
@@ -0,0 +1,145 @@
+)
+@pytest.mark.gpu
+@pytest.mark.newsrec
+def test_prepare_hparams(must_exist_attributes, tmp):
+    data_path = tmp
+    yaml_file = os.path.join(data_path, 'nrms.yaml')
+    train_file = os.path.join(data_path, 'train.txt')
+    valid_file = os.path.join(data_path, 'test.txt')
+    wordEmb_file = os.path.join(data_path, 'embedding.npy')
+
+    if not os.path.exists(yaml_file):
+        download_deeprec_resources(
+            "https://recodatasets.blob.core.windows.net/newsrec/",
+            data_path,
+            "nrms.zip",
+        )
+
+    hparams = prepare_hparams(yaml_file, wordEmb_file=wordEmb_file, epochs=1)
+    assert hasattr(hparams, must_exist_attributes)
+
+@pytest.mark.gpu
+@pytest.mark.newsrec
+def test_load_yaml_file(tmp):
+    data_path = tmp
+    yaml_file = os.path.join(data_path, 'nrms.yaml')
+
+    if not os.path.exists(yaml_file):
+        download_deeprec_resources(
+            "https://recodatasets.blob.core.windows.net/newsrec/",
+            data_path,
+            "nrms.zip",
+        )
+    config = load_yaml(yaml_file)
+    assert config is not None
+
+@pytest.mark.gpu
+@pytest.mark.newsrec
+def test_News_iterator(tmp):
+    data_path = tmp
+    yaml_file = os.path.join(data_path, 'nrms.yaml')
+    train_file = os.path.join(data_path, 'train.txt')
+    valid_file = os.path.join(data_path, 'test.txt')
+    wordEmb_file = os.path.join(data_path, 'embedding.npy')
+
+    if not os.path.exists(yaml_file):
+        download_deeprec_resources(
+            "https://recodatasets.blob.core.windows.net/newsrec/",
+            data_path,
+            "nrms.zip",
+        )
+    
+    hparams = prepare_hparams(yaml_file, wordEmb_file=wordEmb_file, epochs=1)
+    train_iterator = NewsIterator(hparams, hparams.npratio)
+    test_iterator = NewsIterator(hparams, 0)
+
+    assert train_iterator is not None
+    for res in train_iterator.load_data_from_file(train_file):
+        assert isinstance(res, dict)
+        assert len(res) == 5
+        assert res["impression_index_batch"].shape == (hparams.batch_size, 1)
+        assert res["user_index_batch"].shape == (hparams.batch_size, 1)
+        assert res["clicked_news_batch"].shape == (hparams.batch_size, hparams.his_size, hparams.doc_size)
+        assert res["candidate_news_batch"].shape == (hparams.batch_size, hparams.npratio+1, hparams.doc_size)
+
+    assert test_iterator is not None
+    for res in test_iterator.load_data_from_file(valid_file):
+        assert isinstance(res, dict)
+        assert len(res) == 5
+        assert res["impression_index_batch"].shape == (hparams.batch_size, 1)
+        assert res["user_index_batch"].shape == (hparams.batch_size, 1)
+        assert res["clicked_news_batch"].shape == (hparams.batch_size, hparams.his_size, hparams.doc_size)
+        assert res["candidate_news_batch"].shape == (hparams.batch_size, 1, hparams.doc_size)
+
+
+
+@pytest.mark.gpu
+@pytest.mark.newsrec
+def test_NAML_iterator(tmp):
```

Comment:
would it be possible to follow a non-capitalize format for names? `test_NAML_iterator` would be `test_naml_iterator`